source/agora/test/Base.d

```diff
@@ -183,10 +188,14 @@ public final class TestNode (Net) : Node!Net, TestAPI
     /// The metadata used for the tests
     MemMetadata mem_metadata;
 
+    /// The ledger used for the tests
+    Ledger ledger;
+
     ///
     public this (Config config)
     {
         this.mem_metadata = new MemMetadata;
+        this.ledger = new Ledger;
```

Comment:
That code doesn't make sense to me. You instantiate a new `Ledger` just to override it with itself in the parent constructor. You also shadow the parent `ledger` variable ?

Reply:
The shadowing was an oversight. I'll simplify this a bit tomorrow.